pkg/sessionctx/variable/sysvar.go

```diff
@@ -1017,6 +1017,14 @@ var defaultSysVars = []*SysVar{
 			return normalizedValue, nil
 		},
 	},
+	{Scope: vardef.ScopeGlobal, Name: vardef.MaxUserConnections, Value: strconv.FormatUint(vardef.DefMaxUserConnections, 10), Type: vardef.TypeUnsigned, MinValue: 0, MaxValue: 100000,
+		SetGlobal: func(_ context.Context, s *SessionVars, val string) error {
+			vardef.MaxUserConnectionsCount.Store(uint32(TidbOptInt64(val, vardef.DefMaxUserConnections)))
```

Comment:
add a `Validation` for range check

Reply:
> add a `Validation` for range check

Maybe don't need ?
if `set global max_user_connections=-1`, it will be set 0; (min value)
if `set global max_user_connections=100001`, it will be set 100000; (max value)

Reply:
what's the value of `vardef.MaxUserConnectionsCount`?

Reply:
> what's the value of `vardef.MaxUserConnectionsCount`?

It can be set to the maximum of 100000 by the global variable `MAX_USER_CONNECTIONS`